test/transcoding/unused-sret-opaque-ptr.ll

```diff
@@ -6,14 +6,37 @@
 ; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
 ; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
 
-; CHECK-SPIRV-DAG: Name [[#Fun:]] "_Z3booi"
-; CHECK-SPIRV-DAG: Decorate [[#Param:]] FuncParamAttr 3
-; CHECK-SPIRV-DAG: TypePointer [[#PtrTy:]] [[#]] [[#StructTy:]]
+; RUN: llvm-spirv %t.bc -spirv-text -o %t.txt --spirv-ext=+SPV_KHR_untyped_pointers
+; RUN: FileCheck < %t.txt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc -o %t.spv --spirv-ext=+SPV_KHR_untyped_pointers
+; RUNx: spirv-val %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV-DAG: Name [[#FunBoo:]] "boo"
+; CHECK-SPIRV-DAG: Name [[#FunBaz:]] "baz"
+; CHECK-SPIRV-DAG: Decorate [[#ParamBoo:]] FuncParamAttr 3
+; CHECK-SPIRV-DAG: Decorate [[#ParamBaz:]] FuncParamAttr 3
+; CHECK-SPIRV-DAG: TypePointer [[#PtrTy7:]] 7 [[#StructTy:]]
+; CHECK-SPIRV-DAG: TypePointer [[#PtrTy8:]] 8 [[#StructTy:]]
 ; CHECK-SPIRV-DAG: TypeStruct [[#StructTy]]
-; CHECK-SPIRV: Function [[#]] [[#Fun]]
-; CHECK-SPIRV: FunctionParameter [[#PtrTy:]] [[#Param]]
 
-; CHECK-LLVM: call spir_func void @_Z3booi(ptr sret(%struct.Example) align 8
+; CHECK-SPIRV-DAG: TypeFunction [[#BooTy:]] [[#]] [[#PtrTy7]] [[#]] {{$}}
+; CHECK-SPIRV-DAG: TypeFunction [[#BazTy:]] [[#]] [[#PtrTy8]] {{$}}
+
+; CHECK-SPIRV: Function [[#]] [[#FunBoo]] [[#]] [[#BooTy]]
+; CHECK-SPIRV: FunctionParameter [[#PtrTy7]] [[#ParamBoo]]
+
+; CHECK-SPIRV: Function [[#]] [[#FunBaz]] [[#]] [[#BazTy]]
+; CHECK-SPIRV: FunctionParameter [[#PtrTy8]] [[#ParamBaz]]
+
+; CHECK-SPIRV: FunctionParameter [[#PtrTy7]] [[#ParamBar:]]
+; With untyped extension enabled addrspacecast is done to untyped pointer type in addrspace 8.
+; CHECK-SPIRV: PtrCastToGeneric [[#]] [[#Cast:]] [[#ParamBar]]
```

Comment:
If this test now checks more than just unused sret ptr cases, maybe be it worth adding more `CHECK-LLVM` checks corresponding to `CHECK-SPIRV` lines?

Reply:
It actually checks the same, just in more extensive style and with the additional test case - the main checks for SPIR-V still include 1) verification of function attribute and 2) validity of SPV when LLVM IR contains opaque pointers.
As for the CHECK-LLVM part, there is nothing more to check as I see.